### PR TITLE
[#3049] [#3047] va-dropdown rewritten to render reimplementation

### PR DESCRIPTION
**Pull request by aluarius**
Closes: #3049 #3047 

## Description
[x] `va-dropdown` reimplementation to render function.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Comment by m0ksem** [comment]
@aluarius, I fixed VaDropdown render. There are still weird issue with Nuxt. There are hydration mismatch SOMETIMES (for example, in our landing, but not in sandbox). It renders COMMENT node with `v-if` content on server, but on client. I have not clue how to debug it correctly and how we can fix it.

Looks like it is because VaIcon has slot in dynamic tag and there is v-if inside. The last commit here is me fixing hydration missmatch. 

😭

BTW, overall looks good.

File: packages/docs/page-config/ui-elements/dropdown/examples/PlacementAndOffset.vue [hunk 1]
```diff
@@ -22,7 +22,7 @@
             :offset="offset"
           >
             <template #anchor>
-              <Coordinates :placement="placement" />
+              <div><Coordinates :placement="placement" /></div>
```
Comment (m0ksem) [inline packages/docs/page-config/ui-elements/dropdown/examples/PlacementAndOffset.vue:25]: Wrappers can be removed now

File: packages/ui/src/components/va-dropdown/VaDropdown.vue [hunk 2]
```diff
@@ -136,7 +119,11 @@ export default defineComponent({
       if (close && props.trigger !== 'none') { valueComputed.value = false }
     }
 
-    const elRef = useHTMLElement('anchorRef')
+    // const anchorRef = shallowRef<HTMLElement>()
+    const elRef = useHTMLElement('computedAnchorRef')
+    // watch(elRef, (v) => {
```
Comment (m0ksem) [inline packages/ui/src/components/va-dropdown/VaDropdown.vue:124]: This can be removed

**Review by m0ksem — APPROVED** [review]

**Comment by m0ksem** [comment]
The main problem here is to find out how to reproduce this bug in vue sfc to make a bug report.

**Comment by asvae** [comment]
> The main problem here is to find out how to reproduce this bug in vue sfc to make a bug report.

@m0ksem it only reproduces for compiled component?

**Comment by m0ksem** [comment]
> > The main problem here is to find out how to reproduce this bug in vue sfc to make a bug report.
> 
> @m0ksem it only reproduces for compiled component?

Complete opposite. I'm not able to reproduce it in sandbox, where we use compiled vuestic. It seems to be similar to Vue hydration mismatch in transition component. For some reason on client comment vnode with "v-if" doesn't render, so it is not critical.

File: packages/docs/page-config/ui-elements/dropdown/examples/PlacementAndOffset.vue (hunk 1 above)
Comment (aluarius) [reply packages/docs/page-config/ui-elements/dropdown/examples/PlacementAndOffset.vue:25]: Wrapper is still needed. Without it error is thrown.

**Pull request merged**